### PR TITLE
Part of #18714 & #19526: Replacing deprecated constants

### LINKED ISSUE #18714: Replace deprecated constants with enums
### Description
We are currently in the process of migrating to TypeScript. To help reduce tech debt and move the needle closer to a type strict extension it would be great to replace the deprecated design system constant instances with their enum equivalent for all components.

This is a massive undertaking by itself and creating a single PR would be too large. Smaller PRs can be submitted against this issue to ensure easier review and gradual improvements. **A PR can be as small as 1 file or as large as 3.**

#### Deprecated consts and their enum replacements

- `DISPLAY` => `Display`
- `FLEX_DIRECTION` => `FlexDirection`
- `FLEX_WRAP` => `FlexWrap`
- `BLOCK_SIZES` => `BlockSize`
- `SEVERITIES` => `Severity` 
- `TEXT_ALIGN` => `TextAlign`
- `TEXT_TRANSFORM` => `TextTransform`
- `FONT_WEIGHT` => `FontWeight`
- `OVERFLOW_WRAP` => `OverflowWrap`
- `FONT_STYLE` => `FontStyle`

An example of this would be to swap

```jsx
import {
  TEXT_ALIGN,
  DISPLAY,
  FLEX_DIRECTION,
  FLEX_WRAP,
} from '../../../helpers/constants/design-system';

<Component
  textAlign={TEXT_ALIGN.LEFT}
  display={DISPLAY.FLEX}
  flexDirection={FLEX_DIRECTION.ROW}
  flexWrap={FLEX_WRAP.WRAP}
>
```

with 

```jsx
import {
  TextAlign,
  Display,
  FlexDirection,
  FlexWrap,
} from '../../../helpers/constants/design-system';

<Component
  textAlign={TextAlign.Left}
  display={Display.Flex}
  flexDirection={FlexDirection.Row}
  flexWrap={FlexWrap.Wrap}
>
```

### Technical Details
- Replace constants with their enum equivalent

### Acceptance Criteria 
- PRs contain no more than **3 files** this will ensure the PR is easy to review and merged sooner
- Before / After screenshots of the UI to show no visual regressions
- CI tests for jest, linting, and e2e pass 

### Difficulty: Beginner

Good first issue for: External contributors who are familiar with running the extension locally, have knowledge of Jest tests, linting, and are proficient at searching a code base.

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- This PR partially fixes #18714 and #19526 
- Created story for `InstallError` component
- Replaced deprecated constants in the following folders
  - `ui/components/app/snaps/install-error`
  - `ui/components/app/snaps/snap-authorship-header`
  - `ui/components/app/snaps/snap-avatar`

### Small update to `InstallError` error message component otherwise no visual regressions

Screenshot 2023-07-28 at 11 05 16 AM
Screenshot 2023-07-28 at 11 05 46 AM

###  No visual regressions for `SnapAuthorshipHeader`

Screenshot 2023-07-28 at 11 01 15 AM
Screenshot 2023-07-28 at 11 01 35 AM

### No visual regressions for  `SnapAvatar`

Screenshot 2023-07-28 at 11 02 07 AM
Screenshot 2023-07-28 at 11 02 25 AM

## Manual Testing Steps

- Go to the storybook build of this PR
- Check stories for no visual regressions

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, could you please review this PR?
Thanks!

**Review by georgewrmarshall — DISMISSED** [review]
Nice work, made some updates to be more aligned with existing design system components and updated PR description. Thanks for your contribution

**Review by georgewrmarshall — COMMENTED** [review]
Hey @dhruvv173, I think I introduced some linting issues would you mind fixing them please and I'll reach out to the snaps team for a review

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, I think I introduced some linting issues would you mind fixing them please and I'll reach out to the snaps team for a review

The lint issues are fixed, please take a look when possible.

**Review by georgewrmarshall — DISMISSED** [review]
Nice one! LGMT 💯 Thanks for your contribution

**Review by garrettbear — DISMISSED** [review]
Awesome! Thank you 👏

**Review by FrederikBolding — DISMISSED** [review]

File: ui/components/app/snaps/install-error/install-error.js [hunk 1]
```diff
@@ -1,59 +1,31 @@
...
+  Severity,
   TextVariant,
 } from '../../../../helpers/constants/design-system';
-import ActionableMessage from '../../../ui/actionable-message/actionable-message';
-import { AvatarIcon, IconSize, Text } from '../../../component-library';
 
-const InstallError = ({ title, error, description, iconName }) => {
+import { Text, BannerAlert } from '../../../component-library';
+
+const InstallError = ({ title, error, description }) => {
```
[the remark this answers is not in the document]
Comment (eriknson) [reply ui/components/app/snaps/install-error/install-error.js:10]: I'm not sure we should use the banner component to display the main content. Perhaps only for the error message and keep the screen's status icon, title, and description as is.

snap-fetch-error

[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/app/snaps/install-error/install-error.js:10]: That was my bad! I over zealous with some design system component standardization while missing a huge amount of context. Apologies! I completely agree. Your suggestion looks great @eriknson. I'll work with @dhruvv173 to update that

**Comment by dhruvv173** [comment]
hey @georgewrmarshall do we revert the changes mentioned [here](https://github.com/MetaMask/metamask-extension/pull/19562#discussion_r1234962408) so that this can be ready for review?

File: ui/components/app/snaps/install-error/install-error.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/app/snaps/install-error/install-error.js]: Thoughts on this? @Akatori-Design mentioned that the BannerAlert is not needed as the icon and title already state that it's an error message. 

Screenshot 2023-06-23 at 3 14 55 PM

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, can you please update the screenshots in the PR description to latest and we can get this reviewed by the snaps team

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, can you please update the screenshots in the PR description to latest and we can get this reviewed by the snaps team

done!

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for your contribution @dhruvv173

File: ui/components/app/snaps/install-error/install-error.js [hunk 2]
```diff
@@ -36,14 +44,16 @@ const InstallError = ({ title, error, description, iconName }) => {
...
+        <BannerAlert
+          marginTop={4}
+          startAccessory={null}
+          severity={Severity.Danger}
+        >
+          <Text variant={TextVariant.bodySm}>{error}</Text>
+        </BannerAlert>
       )}
     </Box>
   );
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/install-error/install-error.js:59]: Screenshot 2023-07-28 at 11 05 16 AM
Screenshot 2023-07-28 at 11 05 46 AM

File: ui/components/app/snaps/snap-authorship-header/snap-authorship-header.js [hunk 3]
```diff
@@ -64,8 +63,8 @@ const SnapAuthorshipHeader = ({
       <Box
         marginLeft={4}
         marginRight={4}
-        display={DISPLAY.FLEX}
-        flexDirection={FLEX_DIRECTION.COLUMN}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
         style={{ overflow: 'hidden' }}
       >
         <Text ellipsis fontWeight={FontWeight.Medium}>
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-authorship-header/snap-authorship-header.js:70]: Screenshot 2023-07-28 at 11 01 15 AM
Screenshot 2023-07-28 at 11 01 35 AM

File: ui/components/app/snaps/snap-avatar/snap-avatar.js [hunk 4]
```diff
@@ -62,7 +62,7 @@ const SnapAvatar = ({
       ) : (
         <AvatarBase
           size={avatarSize}
-          display={DISPLAY.FLEX}
+          display={Display.Flex}
           alignItems={AlignItems.center}
           justifyContent={JustifyContent.center}
           color={TextColor.textAlternative}
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-avatar/snap-avatar.js:68]: Screenshot 2023-07-28 at 11 02 07 AM
Screenshot 2023-07-28 at 11 02 25 AM

**Review by georgewrmarshall — APPROVED** [review]

**Review by garrettbear — APPROVED** [review]
LGTM

**Review by Mrtenz — APPROVED** [review]

**Comment by georgewrmarshall** [comment]
Looks like there are some e2e tests failing here @dhruvv173

**Pull request merged**